packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -196,28 +196,11 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     if (firstChild == null) {
       if (!addInitialChild(index: firstIndex, layoutOffset: indexToLayoutOffset(itemExtent, firstIndex))) {
         // There are either no children, or we are past the end of all our children.
-        // If it is the latter, we will need to find the first available child.
-        double max;
-        if (childManager.childCount != null) {
-          max = computeMaxScrollOffset(constraints, itemExtent);
-          // TODO(ianh): null-aware flow analysis flags the next two
-          // branches as entirely dead code, and it's hard to argue with
-          // its logic.
-        } else if (firstIndex <= 0) { // ignore: dead_code
+        late double max;
```

Comment:
Does this have to be late? It's assigned in both branches. Should not be needed.

Reply:
Right! And moreover it could be made final.

It would be great to have an hint to detect unnecessary `late` like this one.

Reply:
Actually `final` is not yet accepted by CFE (unlike analyzer)

I just remove `late`.

Reply:
The request to hint/lint unncessary late is tracked at dart-lang/linter#2205